tree.go

```diff
@@ -411,6 +418,21 @@ walk: // Outer loop for walking the tree
 				idxc := path[0]
 				for i, c := range []byte(n.indices) {
 					if c == idxc {
+						if strings.HasPrefix(n.children[len(n.children)-1].path, ":") {
```

Comment:
I think you want to check if the last child is of type Param, right?

Reply:
and then when you need to backtrack, then you can go to the Param child directly?

Reply:
So I think is needed to save the Param child to a place? Otherwise, I will not be able to find it because the walk loop will replace the previous node with the current.